bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/discovery/GreeAirDevice.java

```diff
@@ -0,0 +1,687 @@
+
+        executeCommand(clientSocket, parameters);
+    }
+
+    public void setDeviceHealth(DatagramSocket clientSocket, Integer value) throws GreeException {
+        // Only allow this to happen if this device has been bound
+        if (getIsBound() != Boolean.TRUE) {
+            return;
+        }
+
+        // Set the values in the HashMap
+        HashMap<String, Integer> parameters = new HashMap<>();
+        parameters.put("Health", value);
+
+        executeCommand(clientSocket, parameters);
+    }
+
+    public void setDevicePwrSaving(DatagramSocket clientSocket, Integer value) throws GreeException {
+        // Only allow this to happen if this device has been bound
+        if (getIsBound() != Boolean.TRUE) {
+            return;
+        }
+
+        // Set the values in the HashMap
+        HashMap<String, Integer> parameters = new HashMap<>();
+        parameters.put("SvSt", value);
+        parameters.put("WdSpd", new Integer(0));
+        parameters.put("Quiet", new Integer(0));
+        parameters.put("Tur", new Integer(0));
+        parameters.put("SwhSlp", new Integer(0));
+        parameters.put("SlpMod", new Integer(0));
+
+        executeCommand(clientSocket, parameters);
+    }
+
+    public int getIntStatusVal(String valueName) {
+        /*
+         * Note : Values can be:
+         * "Pow": Power (0 or 1)
+         * "Mod": Mode: Auto: 0, Cool: 1, Dry: 2, Fan: 3, Heat: 4
+         * "SetTem": Requested Temperature
+         * "WdSpd": Fan Speed : Low:1, Medium Low:2, Medium :3, Medium High :4, High :5
+         * "Air": Air Mode Enabled
+         * "Blo": Dry
+         * "Health": Health
+         * "SwhSlp": Sleep
+         * "SlpMod": ???
+         * "Lig": Light On
+         * "SwingLfRig": Swing Left Right
+         * "SwUpDn": Swing Up Down: // Ceiling:0, Upwards : 10, Downwards : 11, Full range : 1
+         * "Quiet": Quiet mode
+         * "Tur": Turbo
+         * "StHt": 0,
+         * "TemUn": Temperature unit, 0 for Celsius, 1 for Fahrenheit
+         * "HeatCoolType"
+         * "TemRec": (0 or 1), Send with SetTem, when TemUn==1, distinguishes between upper and lower integer Fahrenheit
+         * temp
+         * "SvSt": Power Saving
+         */
+        // Find the valueName in the Returned Status object
+        String columns[] = statusResponseGson.packJson.cols;
+        Integer values[] = statusResponseGson.packJson.dat;
+        List<String> colList = new ArrayList<>(Arrays.asList(columns));
+        List<Integer> valList = new ArrayList<>(Arrays.asList(values));
+        int valueArrayposition = colList.indexOf(valueName);
+        if (valueArrayposition == -1) {
+            return -1;
+        }
+
+        // Now get the Corresponding value
+        Integer value = valList.get(valueArrayposition);
+        return value;
+    }
+
+    public boolean hasStatusValChanged(String valueName) throws GreeException {
+        if (prevStatusResponsePackGson == null) {
+            return false;

```

Comment:
I think we should update channels when we don't have the previous value (set initial state of items). 
```suggestion
            return true;
```

Reply:
yep, makes sense